app/lib/actions/katello/repository/update.rb

```diff
@@ -42,6 +43,24 @@ def plan(root, repo_params)
                                        content_type: repository.content_type,
                                        label: content.label,
                                        gpg_url: repository.yum_gpg_key_url)
+
+            if root.previous_changes.key?('os_versions')
+              # If Restrict to OS Version is changing _from_ or _to_ 'rhel-6'...
+              if root.previous_changes['os_versions'].any? { |ch| ch.include?('rhel-6') }
+                # ...force regeneration of entitlement certs on RHEL 6.8 and older
+                Rails.logger.info("Looking for RHEL 6.8 and older clients")
+                cp_consumers = ::Katello::Host::ContentFacet
+                  .joins(host: :operatingsystem)
+                  .where("operatingsystems.major = '6' AND operatingsystems.minor < '9' AND operatingsystems.name = 'RedHat'")
```

Comment:
if there's ever a RHEL 6.10 this `<` will break

Reply:
Another option would be to look for hosts that have a subscription manager installed that is older than what works.  That would be more accurate IMO

Reply:
Yeah that would be ideal but how would I know the installed sub-man version for all my hosts?